### PR TITLE
Add npm ci support

### PR DESCRIPTION
**Pull request by barbelity**
- [x] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/build-info) passed. If this feature is not already covered by the tests, I added new tests.
-----

### CODE REVIEW

File: build-info-extractor-npm/src/test/java/org/jfrog/build/extractor/npm/extractor/NpmExtractorTest.java [hunk 1]
```diff
@@ -174,6 +174,49 @@ public void npmInstallTest(Project project, Set<String> expectedDependencies, St
...
+                {Project.B, Project.B.dependencies, "", true},
+                {Project.B, Collections.emptySet(), "--production", false},
+                {Project.C, Project.C.dependencies, "", true},
+                {Project.C, Project.A.dependencies, "--only=production", true}
+        };
+    }
+
+    @SuppressWarnings("unused")
+    @Test(dataProvider = "npmCiProvider")
+    public void npmCiTest(Project project, Set<String> expectedDependencies, String args, boolean packageJsonPath) {
```
Comment (eyalbe4) [inline build-info-extractor-npm/src/test/java/org/jfrog/build/extractor/npm/extractor/NpmExtractorTest.java:191]: This test is very similar to the ```npmInstallTest``` test. Let's reuse some of the code.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/ClientConfigurationFields.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
     String PACKAGE_MANAGER_ARGS = "package.manager.args";
     String PACKAGE_MANAGER_PATH = "package.manager.path"; // Path to package-manager execution dir
     String PACKAGE_MANAGER_MODULE = "package.manager.module"; // Custom module name for the build-info
+    String NPM_IS_CI_COMMAND = "npm.is.ci.command";
```
[the remark this answers is not in the document]
Comment (barbelity) [reply build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/ClientConfigurationFields.java:36]: Changed and added a description.

File: build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/ArtifactoryClientConfiguration.java [hunk 3]
```diff
@@ -502,8 +504,21 @@ public void setModule(String packageManagerModule) {
...
+    public class NpmHandler extends PrefixPropertyHandler {
+        public NpmHandler() {
+            super(root, PROP_NPM_PREFIX);
+        }
+
+        public boolean isCiCommand() {
+            return rootConfig.getBooleanValue(NPM_IS_CI_COMMAND, false);
+        }
 
+        public void setIsCiCommand(boolean isCiCommand) {
```
[the remark this answers is not in the document]
Comment (barbelity) [reply build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/ArtifactoryClientConfiguration.java:516]: Changed.

File: build-info-extractor-npm/src/test/java/org/jfrog/build/extractor/npm/extractor/NpmExtractorTest.java (hunk 1 above)
Comment (barbelity) [reply build-info-extractor-npm/src/test/java/org/jfrog/build/extractor/npm/extractor/NpmExtractorTest.java:191]: Changed both tests to call a shared method which executes the actual build and validates it.

**Pull request merged**